ambry-protocol/src/main/java/com.github.ambry.protocol/DeleteRequest.java

```diff
@@ -26,28 +27,76 @@
  * Delete request to delete blob
  */
 public class DeleteRequest extends RequestOrResponse {
-  private BlobId blobId;
+  private final BlobId blobId;
+  private final short accountId;
+  private final short containerId;
+  private final long deletionTimeInMs;
   private int sizeSent;
-  private static final short Delete_Request_Version_V1 = 1;
+  private final short version;
+  static final short Delete_Request_Version_1 = 1;
+  static final short Delete_Request_Version_2 = 2;
+  private final static short currentVersion = Delete_Request_Version_1;
 
+  private static final int AccountId_ContainerId_Field_Size_InBytes = 2;
+  private static final int DeletionTime_Field_Size_InBytes = 8;
+
+  /**
+   * Constructs {@link DeleteRequest} in {@link #Delete_Request_Version_1}
+   * @param correlationId correlationId of the delete request
+   * @param clientId clientId of the delete request
+   * @param blobId blobId of the delete request
+   */
+  // @TODO: remove this constructor once DeleteRequest V2 is enabled
   public DeleteRequest(int correlationId, String clientId, BlobId blobId) {
-    super(RequestOrResponseType.DeleteRequest, Delete_Request_Version_V1, correlationId, clientId);
+    this(correlationId, clientId, blobId, MessageInfo.ACCOUNT_ID_LEGACY_VALUE, MessageInfo.CONTAINER_ID_LEGACY_VALUE,
+        (int) Utils.Infinite_Time, currentVersion);
+  }
+
+  /**
+   * Constructs {@link DeleteRequest} in {@link #Delete_Request_Version_2}
+   * @param correlationId correlationId of the delete request
+   * @param clientId clientId of the delete request
+   * @param blobId blobId of the delete request
+   * @param accountId accountId of the blobId being requested
+   * @param containerId containerId of the blobId being requested
+   * @param deletionTimeInMs deletion time of the blob in ms
+   */
+  public DeleteRequest(int correlationId, String clientId, BlobId blobId, short accountId, short containerId,
+      long deletionTimeInMs) {
+    this(correlationId, clientId, blobId, accountId, containerId, deletionTimeInMs, Delete_Request_Version_2);
```

Comment:
this should be `currentVersion` too. Lets not determine versions based on constructors.